src/components/DeleteAccountModal/DeleteAccountModal.tsx

```diff
@@ -0,0 +1,103 @@
+import { useTranslation } from 'react-i18next';
+import { type SchemaOf, object, string } from 'yup';
+import { useNavigate, useLocation } from 'react-router';
+import { useState } from 'react';
+import { useMutation } from 'react-query';
+
+import PasswordField from '../PasswordField/PasswordField';
+import Button from '../Button/Button';
+
+import styles from './DeleteAccountModal.module.scss';
+
+import type { DeleteAccountFormData } from '#types/account';
+import useForm from '#src/hooks/useForm';
+import { addQueryParam, removeQueryParam } from '#src/utils/location';
+import { deleteAccountData, logout } from '#src/stores/AccountController';
+import Alert from '#components/Alert/Alert';
+
+const DeleteAccountModal = () => {
+  const { t } = useTranslation('user');
+
+  const [enteredPassword, setEnteredPassword] = useState<string>('');
+
+  const deleteAccount = useMutation(deleteAccountData, {
+    onSuccess: async () => {
+      await logout();
+      navigate('/');
+    },
+    onError: () => {
+      setEnteredPassword('');
+    },
+  });
+
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  const validationSchema: SchemaOf<DeleteAccountFormData> = object().shape({
+    password: string().required(t('login.field_required')),
+  });
+  const initialValues: DeleteAccountFormData = { password: '' };
+  const { handleSubmit, handleChange, values, errors } = useForm(
+    initialValues,
+    () => {
+      setEnteredPassword(values.password);
+      navigate(addQueryParam(location, 'u', 'delete-account-confirmation'));
```

Comment:
Not sure if you want to be able to use the back button. I think it ends up in a mismatch between the `u` parameter and the presented modal because the confirmation modal is shown based on the `enteredPassword` state.

```suggestion
      navigate(addQueryParam(location, 'u', 'delete-account-confirmation', { replace: true }));
```